nailgun/entities.py

```diff
@@ -1618,6 +1618,40 @@ def __init__(self, server_config=None, **kwargs):
         super(DockerRegistryContainer, self).__init__(server_config, **kwargs)
 
 
+class ContentCredential(
+        Entity,
+        EntityCreateMixin,
+        EntityDeleteMixin,
+        EntityReadMixin,
+        EntitySearchMixin,
+        EntityUpdateMixin):
+    """A representation of a Content Credential entity."""
+
+    def __init__(self, server_config=None, **kwargs):
+        self._fields = {
+            'content': entity_fields.StringField(required=True),
+            'name': entity_fields.StringField(
+                required=True,
+                str_type='alpha',
+                length=(6, 12),
+                unique=True
+            ),
+            'organization': entity_fields.OneToOneField(
+                Organization,
+                required=True,
+            ),
+            'content_type': entity_fields.StringField(
+                choices=('cert', 'gpg_key'),
```

Comment:
perhaps we should add a default value
```python
default = 'gpg_key'
```

Reply:
I would prefer not to, Katello doesn't have a default (https://github.com/Katello/katello/blob/master/app/controllers/katello/api/v2/content_credentials_controller.rb#L15) I don't think Nailgun should be any different.

Reply:
My suggestion has nothing to do with katello, if nailgun was a new implementation I will 100% agree with you guys, but this implementation will break the tests of entities (product, repository ) related to GPGKey.
but if for example we put that default value we can even inherit PGPKey from ContentCredential
```python
class ContentCredential( 
     ...
     default = 'gpg_key'
    ...

class GPGKey(ContentCredential):
    pass
```
with this nothing will be broken.

Reply:
Other suggestion is to not update Product and Repository entity, and let the warning in PGPKey. That warning is catched by the log system and open an issue in robottelo to make changes to the tests related to PGPKey entity (to be replaced by ContentCredential) and when done we can even delete PGPKey entity.